### PR TITLE
Ignore re2 import, remove sqrl-text-functions/web

### PR DESCRIPTION
**Pull request by meyer**
# Problem

`sqrl-text-functions/web` exists to work around an re2 import issue.

# Solution

I fixed the import in the browser by adding `re2: false` to `config.resolve.alias` in our nextjs config file. While I was at it I also tidied up a few loose types.

# Result

We can import stuff from `sqrl-text-functions` and `npm run build` and `npm run dev` work without issue.

### CODE REVIEW

File: packages/sqrl-text-functions/src/InProcessPatternService.ts [hunk 1]
```diff
@@ -15,11 +15,11 @@ function escapeRegExp(str: string): string {
...
-interface GenericRegExp {
-  lastIndex: number;
-  exec(content: string): string[]
-}
-export type CreateRegExp = (source, flags) => GenericRegExp;
+type GenericRegExp = Pick<RegExp, "lastIndex" | "exec">;
+
+export type CreateRegExp = (
+  ...args: Parameters<typeof RegExp>
+) => GenericRegExp;
```
Comment (meyer) [inline packages/sqrl-text-functions/src/InProcessPatternService.ts:22]: @qix I noticed that `source` and `flags` are implicitly set to `any`. Not sure what your feelings are around `noImplicitAny` and stricter types in general but I think we can probably crank the strictness up a little without making the dev experience a nightmare.

File: packages/sqrl-text-functions/src/InProcessPatternService.ts [hunk 2]
```diff
@@ -61,14 +61,18 @@ export class InProcessPatternService implements PatternService {
...
+      return new RegisteredPattern(
+        this.createRegExp(escapeRegExp(pattern.slice(1, -1)), "g")
+      );
     } else {
       // Do a fuzzy match, this could be improved but for now just search with word boundaries and
       // case insensitive
-      return new RegisteredPattern(this.createRegExp("\\b" + escapeRegExp(pattern) + "\\b", "gi"));
+      return new RegisteredPattern(
+        this.createRegExp("\\b" + escapeRegExp(pattern) + "\\b", "gi")
+      );
```
Comment (meyer) [inline packages/sqrl-text-functions/src/InProcessPatternService.ts:75]: thse changes are all from prettier, no actual runtime changes here

File: packages/sqrl-text-functions/src/InProcessPatternService.ts (hunk 1 above)
Comment (qix) [reply packages/sqrl-text-functions/src/InProcessPatternService.ts:22]: @meyer yeah I am good with stricter types, the implicit any was just a relic from the super old codebase.

**Review by qix — APPROVED** [review]
Ah, didn't realize what I was doing wrong in the webpack configuration.

That said it's still useful to have a version of the `sqrl-text-functions` that doesn't depend on `re2`? maybe?

I'm not sure what the standard is here, but maintaining the `/web` export isn't very hard.

**Comment by meyer** [comment]
> That said it's still useful to have a version of the `sqrl-text-functions` that doesn't depend on `re2`? maybe?

I like the config option route that you’re using here. I think we should be able to clean it up a bit more so the webpack override isn’t necessary. I’ll fool around with it a bit in the near future.

**Pull request merged**